src/components/powerInfoPage/table.tsx

```diff
@@ -0,0 +1,303 @@
+        }),
+      },
+    );
+
+    // get response successfully or not
+    if (response.status === 200) {
+      const tmp = await response.json();
+      setApiData(tmp);
+    }
+  };
+
+  // return data type
+  const dataType = (str: string) => {
+    switch (str) {
+      case 'Consume':
+        return '正常用電';
+      case 'NetLoad':
+        return '總淨負載';
+      case 'ESS':
+        return '儲能系統';
+      case 'EV':
+        return '充電樁';
+      case 'WT':
+        return '風能';
+      case 'PV':
+        return '太陽能';
+      default:
+        return 'error';
+    }
+  };
+
+  // create list
+  const dataList = apiData.data.map((d) => {
+    return (
+      <div>
+        <div
+          className={classNames('powerinfo-table-list-item-container', 'text')}
+        >
+          <div className={classNames('powerinfo-table-list-item-date')}>
+            {d.date}
+          </div>
+          <div className={classNames('powerinfo-table-list-item-time')}>
+            {d.time}
+          </div>
+          <div className={classNames('powerinfo-table-list-item-value')}>
+            {d.power_display}
+          </div>
+          <div
+            className={classNames('powerinfo-table-list-item-type')}
+            style={d.power_display >= 0 ? redText : greenText}
+          >
+            {dataType(d.data_type)}
+          </div>
+          <div className={classNames('powerinfo-table-list-item-url')}>
+            <a href={d.address}>&#60;URL&#62;</a>
+          </div>
+          <div className={classNames('powerinfo-table-list-item-infobox')}>
+            <InfoBox
+              date={d.date}
+              type={d.data_type}
+              time={d.time}
+              value={d.power_display}
+              url={d.address}
+            />
+          </div>
+        </div>
+        <hr className={classNames('powerinfo-table-list-item-line')} />
+      </div>
+    );
+  });
+
+  const selectChangeHandler = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setPage(1);
+    setPerPage(event.target.value);
+  };
+
+  const clickFirstPageHandler = () => {
+    setPage(1);
+  };
+
+  const clickPrevPageHandler = () => {
+    if (page > 1) setPage(page - 1);
+    else setPage(1);
+  };
+
+  const clickNextPageHandler = () => {
+    const lastPage =
+      apiData.totalCount % parseInt(per_page, 10) === 0
+        ? apiData.totalCount / parseInt(per_page, 10)
+        : Math.floor(apiData.totalCount / parseInt(per_page, 10)) + 1;
+    if (page < lastPage) setPage(page + 1);
+    else setPage(lastPage);
+  };
+
+  const clickLastPageHandler = () => {
+    const lastPage =
+      apiData.totalCount % parseInt(per_page, 10) === 0
+        ? apiData.totalCount / parseInt(per_page, 10)
+        : Math.floor(apiData.totalCount / parseInt(per_page, 10)) + 1;
+    setPage(lastPage);
+  };
+
+  const [prevDisabled, setPrevDisabled] = useState(true);
+  const [nextDisabled, setNextDisabled] = useState(false);
```

Comment:
Please move these two line to the top of the functional component with other `useState` variables.

Reply:
Done. Thanks!